src/dbnode/storage/stats/query_stats_default_tracker.go

```diff
@@ -53,27 +60,53 @@ func DefaultQueryStatsTracker(
 		MetricsScope().
 		SubScope("query-stats")
 	return &queryStatsTracker{
-		options:    queryStatsOpts,
-		recentDocs: scope.Gauge("recent-docs-per-block"),
-		totalDocs:  scope.Counter("total-docs-per-block"),
+		options:                   queryStatsOpts,
+		recentDocs:                scope.Gauge("recent-docs-matched"),
+		totalDocs:                 scope.Counter("total-docs-matched"),
+		recentBytesRead:           scope.Gauge("recent-bytes-read"),
+		totalBytesRead:            scope.Counter("total-bytes-read"),
+		recentDocsLimitError:      scope.Tagged(map[string]string{"limit": "docs-matched"}).Counter("error"),
+		recentBytesReadLimitError: scope.Tagged(map[string]string{"limit": "bytes-read"}).Counter("error"),
 	}
 }
 
 func (t *queryStatsTracker) TrackStats(values QueryStatsValues) error {
+	docsMatched := values.DocsMatched
+	bytesRead := values.BytesRead
+
+	// Only update the recent metrics on each reset so
+	// we measure only consistently timed peak values.
+	if docsMatched.Reset {
+		t.recentDocs.Update(float64(docsMatched.Recent))
+	}
+	if bytesRead.Reset {
+		t.recentBytesRead.Update(float64(bytesRead.Recent))
+	}
+
 	// Track stats as metrics.
-	t.recentDocs.Update(float64(values.RecentDocs))
-	t.totalDocs.Inc(values.NewDocs)
+	t.totalDocs.Inc(docsMatched.New)
+	t.totalBytesRead.Inc(bytesRead.New)
```

Comment:
This should probably only be emitted after the checks; will also make the graphs fairly clear that queries are being limited when these metrics hit a plateau

Reply:
Makes sense

Reply:
Actually on second thought - I think we probably want to report these still regardless of limit being applied so we have the most accurate info, and we can use the new limit-error metrics to understand to what degree the limits are hit.

Reply:
What would you think of adding both with something like this?
```golang
bytesReadAttempted: scope.Tagged(map[string]string{"status": "attempted"}).Counter("bytes-read")
bytesRead:                   scope.Tagged(map[string]string{"status": "succeeded"}).Counter("bytes-read")
```
Then we can add a graph for read vs requested and visually determine if we've hit limits easily